_posts/2017-05-14-canvass.md

```diff
@@ -0,0 +1,46 @@
+---
+layout:     post
+title:      A/B Testing in Javascript with Canvass
+date:       2017-05-14 15:45:00
+summary:    A/B testing is a popular method for testing changes. Here's how we do it with the Canvass library.
+author:     adam_blanchard
+image:      test-tube.png
+category:   Software Engineering
+tags:       javascript, open-source, experimentation, abtesting
+---
+A/b testing is a popular method used to identify new changes or features which increase some metric (often the conversion rate in our industry). It works by creating a variant of the website which is served to only a subset of the users. It's performance is then compared to the current version of the website.
+
+There are many services out there which provide a/b testing functionality (for example: [Qubit](http://www.qubit.com/), [Maxymiser](https://www.maxymiser.com/) and [Optimizely](https://www.optimizely.com/)). The easiest way to use most of these services is to include a snippet of javascript at the top of your page. You can then build your variants in their online editor. The service takes care of the rest through the javascript snippet, such as pushing different users to a variant (traffic allocation) and tracking the performance of each (data tracking).
+
+We decided to do things a little differently on the Vegas product by building our a/b tests directly in our codebase. This was for a few reasons:
+
+1. We have quite a complex javascript front-end. Our older codebase uses [React](https://facebook.github.io/react/) and [Fluxible](http://fluxible.io/), and our newer one uses [React](https://facebook.github.io/react/), [Typescript](https://www.typescriptlang.org/) and [Alt](http://alt.js.org/). Building variants in another service caused us issues when it was forcing us to modify the DOM outside of our application. We seemed to be spending more time fiddling around with hacks to make them work with our application rather than building the test itself.
+
+2. We have great engineering standards and a good delivery pipeline which we didn't want to bypass by using a third party service. Instead, we wanted to use good coding standards, build trustworthy and well tested features and a use smooth release process in a/b testing, too.
```

Comment:
Is it worth mentioning security concerns here?  Not specifically *which* security concerns, but at least showing that we thought about it?